app/presenters/details_presenter.rb

```diff
@@ -0,0 +1,47 @@
+require "govspeak"
+
+module Presenters
+  class DetailsPresenter
+    attr_reader :content_item_details
+
+    def initialize(content_item_details)
+      @content_item_details = SymbolizeJSON.symbolize(content_item_details)
+    end
+
+    def details
+      @_details ||=
+        begin
+          content_item_details.inject({}) do |seed, hsh|
+            key, val = hsh
+            seed[key] = append_transformed_govspeak(val)
+            seed
+          end
+        end
+    end
+
+  private
+
+    def append_transformed_govspeak(value)
+      return value if value.is_a?(String)
+      return value unless requires_govspeak_html_transform?(value)
+      govspeak = {
+        content_type: "text/html",
+        content: rendered_govspeak(value),
+      }
+      value + [govspeak]
+    end
+
+    def raw_govspeak(value)
+      value.find { |format| format[:content_type] == "text/govspeak" }[:content]
```

Comment:
no, it will blow up. but this method only gets called if `requires_govspeak_html_transform?` returns `true`, so `.find` should always return something.